### ISSUE TITLE
the "new-component-method" branch might need some keep directives

### ISSUE THREAD

**Issue by Helgiii** [haxeui/haxeui-core#274]
I tested my app with the "new-component-method" branches of haxeui-core and haxeui-openfl
and -dce full caused runtime errors. Without it app compiles and runs fine. I use multiple controls, vh grids, scrolls, options, checkboxes, btns, edit, lists.

So, as Jan suggested, there might be some @:keep missing

**Comment by ianharrigan** [comment]
Yup, looks that way... this is very good to know... thanks for doing the legwork!

Ian

**Comment by ianharrigan** [comment]
Hey, what happens if you use `haxe.ui.containers.Grid` rather than `haxe.ui.components.VGrid`?

To be honest, im not entirely sure why a) those components are there and b) what there original purpose was! 😊

Cheers,
Ian

**Comment by Helgiii** [comment]
TypeError: Cannot read property 'invalidateComponent' of null
    at haxe_ui_components__$Label_TextBehaviour.invalidateData (DataBehaviour.hx:27)
    at haxe_ui_components__$Label_TextBehaviour.set (DataBehaviour.hx:15)
    at haxe_ui_behaviours_Behaviours.set (Behaviours.hx:170)
    at haxe_ui_components_Label.set_text (Component.hx:42)
    at gui_PanelSide.InitUi (PanelSide.hx:315)
    at gui_PanelSide.Init (PanelSide.hx:261)
    at DocumentClass.InitApp (Main.hx:557)
    at DocumentClass.Main [as __class__] (Main.hx:117)
    at new DocumentClass (ApplicationMain.hx:326)
    at Function.ApplicationMain.start (ApplicationMain.hx:230)

by the way its not only with -dce full
but also when building release with -final in haxedevelop for html5 target (flash is ok)

**Comment by gpaluk** [comment]
Hit this issue today also, is there a timeline on it? Seems like a pretty core issue since it prevents a release build. Regards

**Comment by ianharrigan** [comment]
wait... what? Thats a thing? Pretty cool... Id like to clean it up a little so you dont have to do that however, i dont suppose you have a small test app that reproduces the hscript issue do you? Are you using binding?

Cheers,
Ian

**Comment by gpaluk** [comment]
> wait... what? Thats a thing? Pretty cool... Id like to clean it up a little so you dont have to do that however, i dont suppose you have a small test app that reproduces the hscript issue do you? Are you using binding?
> 
> Cheers,
> Ian

Hi Ian,

Actually I had just grabbed your Button example and have the following in my test class:

```
Toolkit.init();

var app = new HaxeUIApp();
app.ready(function() {
    var main:Component = ComponentMacros.buildComponent("assets/ui/main.xml");
    app.addComponent(main);
    app.start();
			
    var button: Button = main.findComponent("normal", Button);
        button.onClick = function(e) {
        trace("Event was called");
        stage.displayState = StageDisplayState.FULL_SCREEN_INTERACTIVE;
    };
});
```

**Comment by ianharrigan** [comment]
Nice one - thanks!
